hazelcast/src/main/java/com/hazelcast/map/impl/query/PartitionScanRunner.java

```diff
@@ -97,10 +95,10 @@ public PartitionScanRunner(MapServiceContext mapServiceContext) {
             QueryableEntry queryEntry = new LazyMapEntry(key, value, serializationService, extractors);
```

Comment:
_Minor:_ 
`LazyMapEntry#init` can be used not to create unneeded temporary `LazyMapEntry` objects. When `QueryableEntry` is an appropriate entry for the query then we can create a new `LazyMapEntry` in order to add `result`.

Reply:
Hmm, good idea, but for aggregations it's tricky to determine is the entry was stored somewhere inside the aggregator or not. Aggregators are public API, so we can't change the signature to introduce some kind of a flag, but we can introduce some kind of a marker interface...

Reply:
if we pass this [if](https://github.com/hazelcast/hazelcast/pull/12684/files#diff-3543508995db7e5ead31727a52c5f875L99), we can construct a new `LazyMapEntry` object. This seemed a local change, didn't get the issue with aggregators.

Reply:
Got it, I was thinking about the possibility of reusing the same entry after the `result.add(queryEntry)` call.